pkg/apis/componentconfig/types.go

```diff
@@ -24,6 +24,10 @@ type KubeProxyConfiguration struct {
 	// bindAddress is the IP address for the proxy server to serve on (set to 0.0.0.0
 	// for all interfaces)
 	BindAddress string `json:"bindAddress"`
+	// clusterCIDR is the CIDR range of the pods in the cluster. It is used to
+	// bridge traffic coming from outside of the cluster. If not provided,
+	// no off-cluster bridging will be performed.
+	ClusterCIDR string `json:"clusterCIDR"`
```

Comment:
To be clear, what I'm concerned about is that today we allocate a single consecutive /14 or /16 range for a cluster, which this code covers. But if we want a cluster to be able to have multiple /14s (e.g. for >1k nodes where a user doesn't have consecutive ip space available) then this is going to be fragile and we are either going to break compatibility (by changing this type) or introduce another variable (which is really ugly).